core/src/main/java/org/apache/carbondata/core/datastore/columnar/ByteArrayColumnWithRowId.java

```diff
@@ -42,7 +42,7 @@ public short getRowId() {
   @Override
   public int compareTo(ByteArrayColumnWithRowId o) {
     return UnsafeComparer.INSTANCE
-        .compareTo(column, 2, column.length - 2, o.column, 2, o.column.length - 2);
+        .compareTo(column, 0, column.length, o.column, 0, o.column.length);
```

Comment:
I can see that they were ignoring first two bytes before as it was LV and L is short length field. 
So, now you have changed it to compare whole array. that means it is not coming as LV now ? 
which PR has induced it ?

Reply:
This ByteArrayColumnWithRowId is common for other flows also. please check the impact. test cases may not fail as it is internal logic.

Reply:
As discussed, I think no dictionary string column comparison will go wrong after this change. no dictionary string to be alphabetically sorted we need to ignore L in LV. so please test and handle compare logic based on the flag.

Reply:
yes, this flow will be used for string types only. I have added a check for the dictionary and handled the compare logic.